src/goto-instrument/code_contracts.cpp

```diff
@@ -282,9 +354,345 @@ const symbolt &code_contractst::new_tmp_symbol(
     symbol_table);
 }
 
+exprt create_alias_expression(
+  const exprt &assigns,
+  const exprt &lhs,
+  std::vector<exprt> &aliasable_references)
+{
+  bool first_iter = true;
+  exprt running = false_exprt();
+  for(auto aliasble : aliasable_references)
+  {
+    exprt leftPtr = exprt(ID_address_of, pointer_type(lhs.type()), {lhs});
+    exprt rightPtr = aliasble;
+
+    if(first_iter)
+    {
+      running = same_object(leftPtr, rightPtr);
+      first_iter = false;
+    }
+    else
+    {
+      exprt same = same_object(leftPtr, rightPtr);
+      running = binary_exprt(running, ID_or, same);
```

Comment:
Good point; I believe this would result in a more balanced tree. We will make this change in a later PR expanding this functionality for arrays. The reason for doing it this way was just to handle the case where there are 0 or 1 disjuncts without using additional false expressions, but this can be changed.

Reply:
Done.